File: 7/java/Prime.java
public class Prime {

    public static Boolean isPrime(long x)
    {
        long divisor = 2;

        if (x < 2) {
            return false;
        }

        while (divisor * divisor <= x) {
            if (x % divisor == 0) {
                return false;
            }
            divisor += 1;
        }

        return true;
    }

    public static long getPrime(long n)
    {
        long x = 2;

        while (n > 0) {
            if (isPrime(x)) {
                n -= 1;
            }

            x += 1;
        }

        return x - 1;
    }

    public static void main (String[] args)
    {
        Integer n = Integer.valueOf(args[0]);
        long prime;

        long start = System.nanoTime();
        prime = getPrime(n);
        long end = System.nanoTime();
        long microseconds = (end - start) / 1000;

        Integer primeI = (int) prime;

        System.out.println(n.toString().concat(" prime = ").concat(primeI.toString()));
        System.out.println(microseconds / 1000000.0);
    }
}

File: 7/scala/Prime.scala

object Prime {
//  def isPrime(x: Int, d: Int): Boolean =
//    if (x < 2) false else
//    if (d < 2) true else
//    if (d * d  > x) true else
//    if (x % d > 0) isPrime(x, d + 1) else
//    false

  def isPrime(x: Int, divisor: Int): Boolean = (x, divisor) match {
    case (x, _) if x < 2      => false
    case (_, d) if d < 2      => true
    case (x, d) if d * d > x  => true
    case (x, d) if x % d > 0  => isPrime(x, d + 1)
    case (_, _)               => false
  }

  def countPrimes(x: Int, n: Int): Int = if (n == 0) x - 1  else countPrimes(x + 1, decIfPrime(x, n))

  def decIfPrime(x: Int, n: Int): Int = if (isPrime(x, 2)) n - 1  else n


  def main(args: Array[String]): Unit = {
    val n: Int = args(0).toInt
    val start = System.currentTimeMillis()
    val prime: Int = countPrimes(2, n)
    val time = System.currentTimeMillis() - start

    println(n + " prime = " + prime)
    println(time / 1000.0)
  }
}